### PR TITLE
docs(release): add information on how we tag and release

### LINKED ISSUE #5259: Write up how we want to do tags and feature branches
## Motivation

We want to document the process we decided for new features. We also want to write down the goals we're trying to satisfy.

The draft notes are in the Retrospective pad.
We've also had some conversations in Slack about how this could work.

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

We want to document the process we decided for audits and new features. We also want to write down the goals we're trying to satisfy.

Closes #5259

## Solution

Improve the documentation from our [Zebra versioning and releases](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/release-process.md) 

## Review

This is a draft for revision. I started a whole document but then I realized I was repeating most of the information already available in the Release process and the [release checklist](https://github.com/ZcashFoundation/zebra/blob/main/.github/PULL_REQUEST_TEMPLATE/release-checklist.md)

As @arya2 and @teor2345 recently made the first release candidate, they might have more fresh ideas to add here.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: book/src/dev/release-process.md [hunk 1]
```diff
@@ -32,43 +31,54 @@ The pre-release version is denoted by appending a hyphen and a series of dot sep
...
+
+| Tag    | Description |
+|:---    |:---         |
+| latest | The most recent stable version. |
+| beta   | The most recent pre-release version of Zebra for testing. May not always exist. |
+| rc     | The most recent release candidate of Zebra, meant to become a stable version. |
+
+### Feature Flags {#feature-flags}
+
+To keep the `main` branch in a releasable state, breaking changes and experimental features must be gated behind a [feature flag](https://doc.rust-lang.org/cargo/reference/features.html).
```
Comment (teor2345) [inline book/src/dev/release-process.md]: Some breaking changes are needed immediately.

```suggestion
To keep the `main` branch in a releasable state, experimental features must be gated behind a [Rust feature flag](https://doc.rust-lang.org/cargo/reference/features.html).
Breaking changes should also be gated behind a feature flag, unless the team decides they are urgent.
(For example, security fixes which also break backwards compatibility.)
```

File: book/src/dev/release-process.md [hunk 2]
```diff
@@ -109,10 +119,28 @@ To help ensure that you have sufficient time and a clear path to update, this is
...
+| Rust APIs          | The Rust APIs of the Zebra crates are currently unstable and unsupported. Use the `zebrad` commands or JSON-RPCs to interact with Zebra. |
 
 <a id="process"></a>
 
-## Release candidate & release process
+## Release candidate & release process {#release-process}
 
 Our release checklist is available as a template, which defines each step our team needs to follow to create a new pre-release or release, and to also build and push the binaries to the official channels [Release Checklist Template](https://github.com/ZcashFoundation/zebra/blob/main/.github/PULL_REQUEST_TEMPLATE/release-checklist.md).
+
+## Decision making
```
Comment (teor2345) [inline book/src/dev/release-process.md]: This seems like a rough draft analysis, does it belong in a separate document?
Can we add a key that explains what each symbol means, and a description of what the table is used for?

(Let me know if you'd like me to help, I wrote up the table!)

File: book/src/dev/release-process.md [hunk 3]
```diff
@@ -4,14 +4,13 @@ This document contains the practices that we follow to provide you with a leadin
...
 
-## Zebra versioning
+<a id="versioning"></a>
 
-Zebra version numbers indicate the level of changes that are introduced by the release.
-This use of [semantic versioning](https://semver.org/ "Semantic Versioning Specification") helps you understand the potential impact of updating to a new version.
+## Zebra versioning {#zebra-versioning}
 
-Zebra version numbers have three parts: `major.minor.patch`.
+Zebra version numbers indicate the level of changes that are introduced by the release and are composed of three parts: `major.minor.patch`.
```
Comment (teor2345) [inline book/src/dev/release-process.md]:
```suggestion
Zebra version numbers show the impact of the changes in a release. They are composed of three parts: `major.minor.patch`.
```

File: book/src/dev/release-process.md [hunk 4]
```diff
@@ -32,43 +31,54 @@ The pre-release version is denoted by appending a hyphen and a series of dot sep
...
 
+### Distribution tags {#distribution-tags}
+
+Zebras's tagging relates directly to versions published on Docker. We will reference these [Docker Hub distribution tags](https://hub.docker.com/r/zfnd/zebra/tags) throughout:
+
+| Tag    | Description |
+|:---    |:---         |
+| latest | The most recent stable version. |
+| beta   | The most recent pre-release version of Zebra for testing. May not always exist. |
+| rc     | The most recent release candidate of Zebra, meant to become a stable version. |
```
Comment (teor2345) [inline book/src/dev/release-process.md]:
```suggestion
| rc     | The most recent release candidate of Zebra, meant to become a stable version. May not always exist. |
```

**Review by teor2345 — CHANGES_REQUESTED** [review]
Looks good!

We talked about audit tags and audit branches, but they aren't mentioned in this document.
Can you add a short section based on what we decided in the meeting:
- we'll give the latest `rc` tag to the auditors to review
- when we get the audit report, if the auditors want to review a separate branch with just the audit fixes, we'll make a branch from the audit tag, and make fixes there

File: book/src/dev/release-process.md (hunk 2 above)
Comment (gustavovalverde) [reply book/src/dev/release-process.md]: Yes, I'd appreciate your intervention with this part.

Comment (teor2345) [reply book/src/dev/release-process.md]: I decided to remove it, I wasn't quite sure how to turn it into a useful document, without a lot of extra work.

**Comment by teor2345** [comment]
@gustavovalverde did you want help with the audit part, or did you want to do that in a separate PR?

**Comment by gustavovalverde** [comment]
> @gustavovalverde did you want help with the audit part, or did you want to do that in a separate PR?

I created a ticket for the specific audit part https://github.com/ZcashFoundation/zebra/issues/5431

**Comment by teor2345** [comment]
> > @gustavovalverde did you want help with the audit part, or did you want to do that in a separate PR?
> 
> I created a ticket for the specific audit part #5431

Fair enough, I think that will also be partly covered by #5395, which mentions the audit tag and dependencies.

**Review by teor2345 — APPROVED** [review]
Thanks!

**Pull request merged**